### PR TITLE
Add test for User#visible_environments

### PR DESCRIPTION
**Pull request by ezr-ondrej**

### CODE REVIEW

File: test/models/foreman_puppet/user_test.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      test 'should show the list of environments visible as admin user' do
+        # Admin user sees all environments - including the ones without taxonomies
+        env_names = %w[production global_puppetmaster testing] + [environment.name]
+        assert_equal env_names.sort, User.current.visible_environments.sort
+      end
+
+      test 'should show the list of environments visible as inherited admin user' do
+        User.current = FactoryBot.create(:user, usergroups: [FactoryBot.create(:usergroup, admin: true)]).reload
+        env_names = %w[production global_puppetmaster testing] + [environment.name]
+        assert_equal env_names.sort, User.current.visible_environments.sort
```
Comment (timogoebel) [inline test/models/foreman_puppet/user_test.rb:19]:
```suggestion
        assert_same_elements env_names, User.current.visible_environments
```

**Comment by timogoebel** [comment]
Tests are failing.

File: app/models/concerns/foreman_puppet/extensions/user.rb [hunk 2]
```diff
@@ -3,19 +3,25 @@ module Extensions
...
+        def visible_environments
+          authorized_scope = ForemanPuppet::Environment.unscoped.authorized(:view_environments)
+          authorized_scope = authorized_scope
+                             .joins(:taxable_taxonomies)
+                             .where('taxable_taxonomies.taxonomy_id' => taxonomy_ids[:organizations] + taxonomy_ids[:locations])
+          result = authorized_scope.distinct.pluck(:name)
+          if ::User.current.admin?
+            # Admin users can also see Environments that do not have any organization or location, even when
+            # organizations and locations are enabled.
+            untaxed_environments = Environment.unscoped.where.not(id: TaxableTaxonomy.where(taxable_type: 'ForemanPuppet::Environment').distinct.select(:taxable_id)).pluck(:name)
```
[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply app/models/concerns/foreman_puppet/extensions/user.rb:20]: I think the correct one should be picked up (the passing tests proves that) but better be safe :+1:

[the remark this answers is not in the document]
Comment (timogoebel) [reply app/models/concerns/foreman_puppet/extensions/user.rb:20]: Yeah, this makes it explicit so it matches with the code a couple of lines above.

**Review by timogoebel — APPROVED** [review]

**Comment by ezr-ondrej** [comment]
I'm sorry, but I can't make `.or` to work. Adding it as followup, it is better approach anyway, as we want to include this in current foreman, so we should keep the refactorings on minimum.

**Pull request merged**